### PR TITLE
Pass SDL Vulkan Extensions without copying

### PR DESCRIPTION
**Pull request by ilyas-taouaou**
## Description
Am passing SDL Vulkan  surface extensions directly to the user instead of getting a pointer and copying to it, also am using Uint32 instead of unsigned cause Vulkan always use Uint32.
Also the vulkan library being loaded is not actually a requirement at all, and no need for a vulkan helper function
Also I suggest just returning the platform specific extension instead of an array of extensions
cause all platforms share the surface extension, the only thing that change is the second extension

## Existing Issue(s)
https://github.com/libsdl-org/SDL/issues/7926 and https://github.com/libsdl-org/SDL/issues/7854

### CODE REVIEW

File: include/SDL3/SDL_vulkan.h [hunk 1]
```diff
@@ -160,8 +160,7 @@ extern DECLSPEC void SDLCALL SDL_Vulkan_UnloadLibrary(void);
  *
  * \sa SDL_Vulkan_CreateSurface
  */
-extern DECLSPEC SDL_bool SDLCALL SDL_Vulkan_GetInstanceExtensions(unsigned int *pCount,
-                                                                  const char **pNames);
+extern DECLSPEC char const* const* SDLCALL SDL_Vulkan_GetInstanceExtensions(Uint32 *pCount);
```
Comment (madebr) [inline include/SDL3/SDL_vulkan.h:163]: Please update the documentation.

File: test/testvulkan.c [hunk 2]
```diff
@@ -222,36 +222,15 @@ static void createInstance(void)
...
-        SDL_free((void *)extensions);
-        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION,
-                     "SDL_Vulkan_GetInstanceExtensions(): %s\n",
-                     SDL_GetError());
-        quit(2);
-    }
-    instanceCreateInfo.enabledExtensionCount = extensionCount;
-    instanceCreateInfo.ppEnabledExtensionNames = extensions;
+
+    instanceCreateInfo.ppEnabledExtensionNames = SDL_Vulkan_GetInstanceExtensions(&instanceCreateInfo.enabledExtensionCount);
```
Comment (madebr) [inline test/testvulkan.c:232]: You need to free the extension names somewhere.
(run testvulkan with `--trackmem` to track allocations)

Comment (ilyas-taouaou) [reply test/testvulkan.c:232]: the extension names aren't being allocated anymore, they're returned directly

**Comment by ilyas-taouaou** [comment]
Can I return the OS specific extension instead of a list of extensions?

**Comment by ilyas-taouaou** [comment]
> Can I return the OS specific extension instead of a list of extensions?

@madebr

File: test/testvulkan.c (hunk 2 above)
Comment (ilyas-taouaou) [reply test/testvulkan.c:232]: INFO: Memory allocations:
Total: 0.00 Kb in 0 allocations

Comment (madebr) [reply test/testvulkan.c:232]: That's even better :)

Comment (madebr) [reply test/testvulkan.c:232]: Perhaps document that the returned pointer should not be freed.

File: include/SDL3/SDL_vulkan.h [hunk 3]
```diff
@@ -152,16 +152,13 @@ extern DECLSPEC void SDLCALL SDL_Vulkan_UnloadLibrary(void);
  *
  * \param pCount A pointer to an unsigned int corresponding to the number of
  *               extensions to be returned
```
Comment (madebr) [inline include/SDL3/SDL_vulkan.h:154]:
```suggestion
 * \param pCount A pointer to an unsigned int corresponding to the number of
 *               extensions returned
```

File: include/SDL3/SDL_vulkan.h [hunk 4]
```diff
@@ -152,16 +152,13 @@ extern DECLSPEC void SDLCALL SDL_Vulkan_UnloadLibrary(void);
  *
```
Comment (madebr) [inline include/SDL3/SDL_vulkan.h:152]: Also, please review the text above. There is no more `pNames`.

**Comment by madebr** [comment]
> > Can I return the OS specific extension instead of a list of extensions?
> 
> @madebr

@icculus is the man you need to convince :)

**Comment by icculus** [comment]
This is a really nice simplification, great work.  :)

**Comment by icculus** [comment]
(Hmm, that was suppose to squash those commits, oh well.)

**Comment by ilyas-taouaou** [comment]
> This is a really nice simplification, great work. :)

it can be much simpler if we just pass the OS depedent extension only, instead of an array
since the first element is always the same (VK_SURFACE_KHR)

**Comment by sezero** [comment]
Someone should update README-migration and sdl2-compat after this

**Comment by icculus** [comment]
> it can be much simpler if we just pass the OS depedent extension only, instead of an array
since the first element is always the same (VK_SURFACE_KHR)

I'd rather keep it this way, in case some weird three-extension thing comes along later, or we all replace VK_SURFACE_KHR with VK_SURFACE2_KHR or something.

**Comment by icculus** [comment]
> Someone should update README-migration and sdl2-compat after this

I'm doing this now!

**Comment by icculus** [comment]
Documentation is fixed in 07a776f2553245cbbf39642e8876eb1c2701cf79, README-migration.md is updated in 5b3a2c6df66a58f8694ccb917f7f5641cf699b2d.

**Comment by icculus** [comment]
Updated sdl2-compat and confirmed testvulkan still works in https://github.com/libsdl-org/sdl2-compat/commit/15cdc96b81ff6baaa9a514def1f4bb8f948dad18

**Pull request merged**